roles/rsyslog/templates/output_forwards.j2

```diff
@@ -1,18 +1,28 @@
+{% if item.udp_port | d() %}
+{%   set __forwards_port = item.udp_port %}
+{%   set __forwards_protocol = 'udp' %}
+{% elif item.tcp_port | d() %}
+{%   set __forwards_port = item.tcp_port %}
+{%   set __forwards_protocol = 'tcp' %}
+{% else %}
+{%   set __forwards_port = __rsyslog_default_port %}
```

Comment:
```suggestion
{%   set __forwards_port = item.port | d(__rsyslog_default_port) %}
```

Reply:
I just noticed that using the previous playbook the client got default port.

Reply:
??? We are switching from `port: some_number` and `protocol: tcp or udp` to `tcp or udp_port: some_number`, aren't we? Why we still need the `port` key there? Also, the {% else %} clause indicates no `tcp_port` nor `udp_port` was given. In that case, the default protocol is set to `tcp` and the default port is 514. I think this is the same behaviour as before.
```
Per request from Dalibor Pospíšil, let forwards output (client) and remote input (server)
share the same protocol and port key interface.
    
old:
    protocol: {tcp,udp}
    port: port_number
new:
    {tcp,udp}_port: port_number
```

Reply:
Sure, let's remove it completely. But then do not configure omfwd at all as it is safer not to send anything rather than sending messages somewhere unintentionally.

Reply:
Hmm, you are suggesting not to have the default protocol and port settings in omfwd? I.e., no tcp_port nor udp_port were given, the logging role should fail?

Reply:
Probably, client side is in this regard more sensitive. I think this needs more opinions. @richm @jvymazal

Reply:
I'm not sure what the question is

Reply:
> I'm not sure what the question is

Sorry, it was not very clear...
In the configuration for forwards output (omfwd), the current logging role implementation allows not to set port and protocol. I.e., if both tcp_port and udp_port are **not** set in the forwards output, the omfwd is fowarding the logs to the target host's port 514 over udp, which is the default behaviour that rsyslog provides. I think @sopos suggests to check if either tcp_port or udp_port exists in the forwards output config varaibles, and make the logging role fail if both are not set to reduce the unexpected behaviour. (@sopos, is my understanding correct?) Please note that the target host/IP is being checked and if it's missing, the item is skipped. The logging role does not fail, but just the item is ignored.

Reply:
One of the principles of system roles is that we try to use reasonable default values for the user.  So if the user specifies that they want to use omfwd, but does not specify port/protocol, the role should assume that the user does indeed want to use fowarding with the default port and protocol.  So yes - we should not require the user to set the port and protocol.